File: src/test/java/com/tartner/dancehours/domain/danceuser/external/DanceUserAggregateQueryModelTest.java
package com.tartner.dancehours.domain.danceuser.external;

import com.tartner.dancehours.querymodel.danceuser.DanceUserRepository;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class DanceUserAggregateQueryModelTest {

    public static final String TestEmail = "bill@tartner.com";
    private DanceUserRepository dao;
    private DanceUserAggregateQueryModel queryModel;
    @Before
    public void setUp() throws Exception {

        dao = mock(DanceUserRepository.class);
        queryModel = new DanceUserAggregateQueryModel( dao );
    }

    @Test
    public void testEmailAlreadyExistsWhenExists() throws Exception {
        when( dao.existsByEmail( TestEmail ) ).thenReturn( true );

        assertThat( queryModel.emailAlreadyExists( TestEmail ), is( true ) );
    }

    @Test
    public void testEmailAlreadyExistsWhenDoesNotExist() throws Exception {
        when( dao.existsByEmail( TestEmail ) ).thenReturn( false );

        assertThat(queryModel.emailAlreadyExists( TestEmail ), is( false ) );
    }
}

File: src/test/java/com/tartner/dancehours/domain/danceuser/external/DanceUserAggregateQueryModelTest.kt
package com.tartner.dancehours.domain.danceuser.external

import com.tartner.dancehours.querymodel.danceuser.DanceUserRepository
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

public class DanceUserAggregateQueryModelTest {
    private var dao: DanceUserRepository? = null
    private var queryModel: DanceUserAggregateQueryModel? = null

    Before
    public fun setUp() {
        dao = mock(javaClass<DanceUserRepository>())
        queryModel = DanceUserAggregateQueryModel(dao as DanceUserRepository)
    }

    Test
    throws(Exception::class)
    public fun testEmailAlreadyExistsWhenExists() {
        `when`(dao!!.existsByEmail(TestEmail)).thenReturn(1L)

        assertThat(queryModel!!.emailAlreadyExists(TestEmail), `is`(true))
    }

    Test
    throws(Exception::class)
    public fun testEmailAlreadyExistsWhenDoesNotExist() {
        `when`(dao!!.existsByEmail(TestEmail)).thenReturn(0L)

        assertThat(queryModel!!.emailAlreadyExists(TestEmail), `is`(false))
    }

    companion object {

        public val TestEmail: String = "bill@tartner.com"
    }
}
